src/_includes/base/list.macros.njk

```diff
@@ -0,0 +1,26 @@
+{%- macro all(pages, collections) -%}
+  {%- if pages | length > 0 -%}
+    <ul>
+      {%- for page in pages | byDate -%}
+        <li>
+          {%- if page.data.draft -%}
+            <strong> [draft] </strong>
+          {%- endif -%}
+
+          <a href="{{ page.url }}">
+            {{- page.data.title -}}
+          </a>
+
+          {%- set date = collections[page.data.index] | latest
+            if page.data.index
+            else page.date
```

Comment:
It may not make a practical difference, but for consistency should this be:

```suggestion
            else collections.all | getPage(page) | getDate
```

Reply:
- `getPage(page)` isn't needed here. It's only needed when we have a `page` object without the `data` property (at the top level). Since these are page objects from a collection, they already have that. 
- `getDate` also shouldn't be needed here? The earlier `pages | byDate` already runs each page through `upDated`, which gets the date and updates the `page.date` value. 

Or am I missing something?

Reply:
Yeah, I think you're correct -- I just get nervous about tracking when `upDated` has been called in what context.